gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -358,8 +358,8 @@ private void publishFileSet(CopyEntity.DatasetAndPartition datasetAndPartition,
       CopyEntity copyEntity = CopySource.deserializeCopyEntity(wus);
       if (copyEntity instanceof CopyableFile) {
         CopyableFile copyableFile = (CopyableFile) copyEntity;
-        preserveFileAttrInPublisher(copyableFile);
         if (wus.getWorkingState() == WorkingState.COMMITTED) {
+          preserveFileAttrInPublisher(copyableFile);
```

Comment:
I'm glad the fix was a mere reordering.  still, that is so subtle, it definitely deserves a comment (e.g. about what would transpire if preserving before verifying `COMMITTED`)

Reply:
Added a comment